crates/weaver_resolver/src/lib.rs

```diff
@@ -419,10 +466,16 @@ impl SchemaResolver {
         resolver_config: ResolverConfig,
         cache: &Cache,
         log: impl Logger + Clone + Sync,
+        policy_engine: Option<&weaver_checker::Engine>,
```

Comment:
This is honeslty my only contention with this whole PR here.

SHOULD the policy checking be part of the resolver or can this be a separate thing you need to piece together later as a consumer of these crates.

Reply:
I extracted the policy engine from the SchemaResolver and refactored a bit the command line accordingly.